content/en/docs/concepts/security/k8s-hardening-guide.md

```diff
@@ -0,0 +1,112 @@
+- **CRI Files** - Where the Kubernetes control plane components are running as pods in the cluster, a Container Runtime will be running on the control plane (e.g. Docker, ContainerD, CRI-O). It is important to protect access to the programs and configuration of their CRI, in particular access to any socket files used by the CRI (e.g. `docker.sock`, `containerd.sock`) should be closely controlled as unauthorised access to these sockets will allow attackers to compromise the control plane services.
+
+## Worker Node Configuration
+TODO: This is generally going to follow the same path as the Control plane configuration, but focused on the kubelet, kube-proxy, and network plugins. (Ideally this should cover Windows as well as Linux)
+
+## PKI Management
+TODO: Some possible areas for this section would be to look at how cluster owners could securely manage the keys for their certificate authorities. Also looking at the different CAs deployed by a cluster (main, etcd and aggregated API server) and how their separate keys should be managed 
+Plus details like making sure that the kubelet’s TLS listener is using a signed / trusted certificate.
+
+## Cluster Authentication
+TODO: Some areas we could/should cover here are tradeoffs and security concerns for various forms of authentication. Obviously coverage of the in-built mechanisms makes sense (token, client cert) but given that most prod. Clusters will not rely on only the in-built mechanisms, some coverage of the external Authentication options would make sense.
+
+Also seems like this might be a good place to discuss service account token auth, and bootstrap tokens.
+
+A general topic for discussion could include how credential revocation is handled for the in-built options. 
+
+## Authorization
+TODO: Topics that could definitely be handled here are the choices & tradeoffs from the different authorization modes and things like the benefits of node authorization. The main topic of discussion (RBAC) may be a separate document (The RBAC Guide ) .
+
+## Workload Security Configuration
+TODO: There’s a number of topics we might want to cover here. Whilst the technical details of things like Security Context and PSPs/PSS are well covered in existing documents, what could be good is to talk about use cases and tradeoffs and how different options can be applied.
+
+## Network Policy Configuration
+TODO: For this section we could go into common good designs for Network policies in clusters. For example how to set up a default deny ingress/egress policy per namespace and then add individual requirements for applications to communicate inside and outside the cluster. 
+It could also be worth flagging up specific risks that can be addressed by network policies, like container access to cloud metadata services.
+Another topic that we could touch on would be the risks that pods using host networking will bypass network policy controls
```

Comment:
```suggestion
By default all ingress and egress traffic is allowed within a Kubernetes cluster. This can give attackers unnecessary opportunities to pivot, download tools, or egress sensitive data from pods.

[Network Policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) can be used to prevent these types of attack paths. To use network policies, a Container Network Interface (CNI) plugin that supports network policies is required.

You can default deny all ingress and/or egress traffic with a network policy. The policy below will deny all ingress and egress traffic for any pod in the `default` namespace, unless there is another network policy that explicitly allows traffic.

\```yaml
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
    name: default-deny-all
    namespace: default
spec:
    podSelector: {}
    policyType:
    - Ingress
    - Egress
\```

You can then explicitly allow ingress and egress traffic your applications require using additional network policies.

It's best to have default deny policies for all namespaces, but this must be carefully implemented to avoid breaking running services.

If you are running a cluster on a cloud provider, you should also deploy a network policy to prevent pods from accessing cloud metadata APIs by default, e.g.: `169.254.169.254`. This can prevent attackers pivoting from a pod to gain access to cloud credentials a worker node may have.
```

Reply:
I'm debating how deep to go in terms of providing network policy examples. I don't think we want to teach how network policies work, more than explain what kind of policies are important and why. Open to feedback here 🙏